### ISSUE TITLE
Swift 4 support

### ISSUE THREAD

**Issue by leszarna** [RxSwiftCommunity/Action#105]
Hi,

Is this project last repo already extensively tested and passing tests with swift4 ?
I am getting EXC_BAD_ACCESS called at Action:100  : 
`    Observable
            .combineLatest(executing, enabledIf) { !$0 && $1 }
            .bind(to: enabledSubject)
            .disposed(by: disposeBag)` 
in .bind(to...)
at: `override func subscribe<O : ObserverType>(_ observer: O) -> Disposable where O.E == Element {
        if !CurrentThreadScheduler.isScheduleRequired {
            // The returned disposable needs to release all references once it was disposed.
            let disposer = SinkDisposer()
            let sinkAndSubscription = run(observer, cancel: disposer)
            disposer.setSinkAndSubscription(sink: sinkAndSubscription.sink, subscription: sinkAndSubscription.subscription)`
 in disposer.setSink...

Thats my only problem using RxSwift and not sure where is the problem.

**Comment by ashfurrow** [comment]
Hi there! I don't think anyone's tested this for Swift 4 yet (I know I haven't). The solution doesn't appear obvious, I'll try to take a look this weekend and get back to you.

**Comment by fpillet** [comment]
I'm working on a Swift 4 PR. Coming right up.

**Comment by fpillet** [comment]
BTW @leszarna I'm not sure your problem comes from Action itself. Would be nice if you could try and repro once we make the `swift4.0` branch available.

**Comment by leszarna** [comment]
@fpillet I've tried your PR and it works very well for me now.

@fpillet Thank you very much for your help.

Thank you @ashfurrow. 
This is really excellent piece of abstraction so thank you all working this.

Regards
Leszek

**Comment by adamsmaka** [comment]
Hi guys, I'm newbie with cocoapods and things like that :)
Xcode 9 is stable now so we have Swift 4 officially released.  I updated my RxSwift and RxCocoa to use `swift4.0` branch, but how can I do it with Action?
When I start pod install I got errors like this:
`pod install
Analyzing dependencies
[!] Unable to satisfy the following requirements:

- `RxSwift (from `https://github.com/ReactiveX/RxSwift.git`, branch `swift4.0`)` required by `Podfile`
- `RxSwift (= 4.0.0-beta.0)` required by `Podfile.lock`
- `RxSwift (~> 4.0.0-beta.0)` required by `RxCocoa (4.0.0-beta.0)`
- `RxSwift (~> 4.0.0-alpha.0)` required by `RxDataSources (2.0.0-alpha.0)`
- `RxSwift (~> 2.0.0-beta)` required by `Action (0.1.0)`
`

**Comment by ashfurrow** [comment]
Hi @adamsmaka – I'd be happy to help with this. Can you post your Podfile? Or at least the relevant portions of it? It looks like we might need to updated `RxDataSources`'s podspec to point to the new beta: https://github.com/RxSwiftCommunity/RxDataSources/blob/swift4.0/RxDataSources.podspec#L44

**Comment by adamsmaka** [comment]
of course, here it is
```
  pod 'RxSwift', git: 'https://github.com/ReactiveX/RxSwift.git', branch: 'swift4.0'
  pod 'RxCocoa', git: 'https://github.com/ReactiveX/RxSwift.git', branch: 'swift4.0'
  pod 'RxDataSources', git: 'https://github.com/RxSwiftCommunity/RxDataSources.git', branch: 'swift4.0'
  pod 'Action'
```

**Comment by ashfurrow** [comment]
Thanks! I've reproduced the issue and I think it's caused by Action's pre-release not being on CocoaPods trunk. We should probably put Action up on trunk – my machine's setup is... not working right now for Swift 4 😞  @fpillet could you push the current podspec to trunk? 

In the mean time, @adamsmaka , you can use Action's pre-release podspec with the following:

```rb
  pod 'Action', git: 'https://github.com/RxSwiftCommunity/Action.git', branch: 'master'
```

I had some issues with the branch names in your pod file, but was able to get CocoaPods working with the following:

```rb
  pod 'RxSwift', git: 'https://github.com/ReactiveX/RxSwift.git', branch: 'rxswift4.0-swift4.0'
  pod 'RxCocoa', git: 'https://github.com/ReactiveX/RxSwift.git', branch: 'rxswift4.0-swift4.0'
  pod 'RxDataSources', git: 'https://github.com/RxSwiftCommunity/RxDataSources.git', branch: 'swift4.0'
  pod 'Action', git: 'https://github.com/RxSwiftCommunity/Action.git', branch: 'master'
```

**Comment by fpillet** [comment]
I'm currently on travel, in crunch mode and away from my usual Mac so not possible for me before next monday unfortunately

**Comment by adamsmaka** [comment]
ok so we are waiting...

**Comment by ashfurrow** [comment]
No worries, I'll take a look 👍 In the mean time, everything works with the podfile I posted above.

**Comment by bobgodwinx** [comment]
This is already fixed in #106
